client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -1144,21 +1226,19 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
     masterAndSlavePairs._1.add(masterLocation)
   }
 
-  private def reallocateForNonExistPartitionLocationFromCandidates(
-      partitionId: Int,
-      oldEpochId: Int,
-      candidates: List[WorkerInfo]): WorkerResource = {
-    val slots = new WorkerResource()
-    allocateFromCandidates(partitionId, oldEpochId, candidates, slots)
-    slots
-  }
-
+  /**
+   * Reallocate slots from available workers.
+   *
+   * @param partitionIdToEpochs The list of origin partition's id and epoch number.
+   * @param candidates Available worker.
+   * @return Reallocated PartitionLocations.
+   */
   private def reallocateSlotsFromCandidates(
-      oldPartitions: List[PartitionLocation],
+      partitionIdToEpochs: Array[(Int, Int)],
```

Comment:
unnecessary change? but it's good to use array or scala collection so we can avoid converting.

Reply:
> unnecessary change? but it's good to use array or scala collection so we can avoid converting.

Double check code, seems we don't need 
```
   val partitionIdToEpochs = changePartitions.map { split =>
      if (split.oldPartition != null) {
        split.oldPartition.getId -> split.oldPartition.getEpoch
      } else {
        split.partitionId -> split.epoch
      }
    }
```

This change is unnecessary